### DISCUSSION TITLE
What have been our biggest maintenance pain points?

### DISCUSSION THREAD

**Discussion by with-heart** [statelyai/xstate#4986] [General]
This is pretty specifically targeted at @davidkpiano and @Andarist but I'm wondering what have been your biggest pain points around maintaining the repo. Anything in particular you've wanted to automate/deal with but haven't had the time?

**Comment by Andarist** [comment]
Better type-level tests would be great, being able to test completions and stuff like that

**Reply by with-heart** [reply]
This pairs nicely with David's wish to move to `vitest` since it has built-in type testing stuff

**Comment by davidkpiano** [comment]
As a couple wish-list items:

- I'd love to move the tests over to [Vitest](https://vitest.dev/)
- I'd also like to use [pnpm](https://pnpm.io/) instead of Yarn
- Changesets isn't intuitive for pre-releases, but that's a later problem now

**Reply by with-heart** [reply]
Your wishlist is also my wishlist 🧞
